server/core/tests/test_services.py

```
@@ -0,0 +1,115 @@
+        }
+        
+        # This should actually raise KeyError since display_name is accessed directly in create_user
+        with self.assertRaises(ValidationError):
+            UserService.create_user(invalid_data)
+    
+    def test_create_user_with_existing_username(self):
+        """Test creating a user with an existing username."""
+        invalid_data = self.valid_user_data.copy()
+        invalid_data['username'] = 'existinguser'
+        
+        with self.assertRaises(Exception):  # Django will raise an IntegrityError for duplicate username
```

Ruff (0.8.2)

`assertRaises(Exception)` should be considered evil

(B017)